4-domain.go

```diff
@@ -0,0 +1,109 @@
+package domain
+
+import (
+	"github.com/nspcc-dev/neo-go/pkg/interop/runtime"
+	"github.com/nspcc-dev/neo-go/pkg/interop/storage"
+)
+
+// Main is a very useful function.
+func Main(operation string, args []interface{}) interface{} {
+	ctx := storage.GetContext()
+
+	// Queries the domain owner
+	if operation == "query" {
+		if len(args) != 1 {
+			return false
+		}
+
+		domainName := args[0].([]byte)
+		message := "QueryDomain: " + args[0].(string)
+		runtime.Log(message)
+
+		owner := storage.Get(ctx, domainName)
+		if owner != nil {
+			runtime.Notify(owner)
+			return owner
+		}
+
+		runtime.Log("Domain is not yet registered")
+	}
+
+	// Deletes the domain
+	if operation == "delete" {
+		if len(args) != 1 {
+			return false
+		}
+		domainName := args[0].([]byte)
+		message := "DeleteDomain: " + args[0].(string)
+		runtime.Log(message)
+
+		owner := storage.Get(ctx, domainName)
+		if owner == nil {
+			runtime.Log("Domain is not yet registered")
+			return false
+		}
+
+		if !runtime.CheckWitness(owner.([]byte)) {
+			runtime.Log("Sender is not the owner, cannot delete")
+			return false
+		}
+
+		storage.Delete(ctx, domainName)
+		runtime.Notify("Domain is successfully deleted")
```

Comment:
This is a good message for `Log`, but I'd expect something more interesting from notification. I think we should give good examples here in how and when to use logs and how and when to use notifications. Take a look at the NEP5 and its events, it's a good example. Here I'd expect to get some kind of `["deleted", owner, domainName]` notification that would be very useful for any external user of it (think of it as an API).

Reply:
Done, now it is `runtime.Notify("deleted", owner, domainName)`, but in this case we have only `"deleted"` in Notifications and other two values on the stack. Is it desired behaviour? Btw, nep5 acts the same way.

Reply:
Looks like compiler bug to me (not treating varargs properly), please file an issue for it. Meanwhile you can do
```
runtime.Notify([]interface{}{"msg", smth1, smth2})
```